django/db/models/functions/text.py

```diff
@@ -188,6 +191,121 @@ def as_sqlite(self, compiler, connection, **extra_context):
         return super().as_sql(compiler, connection, function='UNICODE', **extra_context)
 
 
+class RegexpFlagMixin:
+    default_flags = {
+        'mariadb': 'c',  # Force case-sensitive matching.
+        'mysql': 'c',  # Force case-sensitive matching.
+        'postgresql': 'p',  # Force partial newline-sensitive matching.
+    }
+    inline_flags = {}
+    wrap_pattern = {}
+
+    @cached_property
+    def flag_mapping(self):
+        return {k: str.maketrans(v) for k, v in {
+            'mariadb': {'c': '-i'},
+            'mysql': {'s': 'n'},
+            'oracle': {'s': 'n'},
+            'postgresql': {'m': 'n'},
+        }.items()}
+
+    def as_sql(self, compiler, connection, **extra_context):
+        expression, pattern, *extra, flags = self.get_source_expressions()
+
+        if connection.vendor == 'mysql' and connection.mysql_is_mariadb:
+            vendor = 'mariadb'
+        else:
+            vendor = connection.vendor
+
+        # Add default flags to ensure consistency between database backends.
+        if vendor in self.default_flags:
+            flags.value = self.default_flags[vendor] + flags.value
```

Comment:
You should create a new `Value` here as you're altering the one assigned to `self` in place so it gets compiled multiple times `self.default_flags[vendor]` will be added multiple times.

```suggestion
        if vendor in self.default_flags:
            flags = Value(self.default_flags[vendor] + flags.value)
```